ipaplatform/debian/tasks.py

```diff
@@ -88,4 +98,99 @@ def configure_pkcs11_modules(self, fstore):
     def restore_pkcs11_modules(self, fstore):
         pass
 
+    def insert_ca_certs_into_systemwide_ca_store(self, ca_certs):
+        result = True
+
+        # pylint: disable=ipa-forbidden-import
+        from ipalib import x509  # FixMe: break import cycle
+        from ipalib.errors import CertificateError
+        # pylint: enable=ipa-forbidden-import
+
+        # TODO: it would be nice to have paths.IPA_P11_KIT created here, so
+        # that it could still be used by p11-kit in the future if Debian were
+        # to adopt Red Hat's CA certificate management code. If we were to do
+        # that, the code should be split out into a separate method that both
+        # DebianTaskNamespace and RedHatTaskNamespace can call.
+
+        ca_certificates_dir_path = Path(paths.CA_CERTIFICATES_DIR)
+        try:
+            ca_certificates_dir_path.mkdir(mode=0o755, exist_ok=True)
+        except OSError as e:
+            logger.error(
+                'Could not create %s: %s', ca_certificates_dir_path, e)
+            return False
+
+        for i, ca_cert in enumerate(ca_certs):
+            cert, nickname, trusted, _ext_key_usage = *ca_cert,
+
+            if not trusted:
+                continue
+
+            try:
+                # XXX is rfc4514_string really displaying the DN
+                # elements in the wrong order?
+                subject = cert.subject.rfc4514_string()
+                issuer = cert.issuer.rfc4514_string()
+                serial_number = cert.serial_number
+            except (PyAsn1Error, ValueError, CertificateError) as e:
+                logger.warning(
+                    "Failed to decode certificate \"%s\": %s", nickname, e)
+                continue
+
+            ca_path = ca_certificates_dir_path / f'ipa-ca-{i}.crt'
+
+            try:
+                with open(ca_path, 'w') as f:
+                    os.fchmod(f.fileno(), 0o644)
+
+                    f.write(f'''\
```

Comment:
style nit-pick: we prefer double triple quotes.

Reply:
OK